modules/store/src/reducer_creator.ts

```diff
@@ -42,11 +40,14 @@ export interface OnReducer<State, Creators extends readonly ActionCreator[]> {
  * ```
  */
 export function on<State, Creators extends readonly ActionCreator[]>(
-  ...args: [...creators: Creators, reducer: OnReducer<State, Creators>]
+  ...args: [
+    ...creators: Creators,
+    reducer: OnReducer<State extends infer S ? S : never, Creators>
+  ]
 ): ReducerTypes<State, Creators> {
   // This could be refactored when TS releases the version with this fix:
   // https://github.com/microsoft/TypeScript/pull/41544
-  const reducer = args.pop() as OnReducer<State, Creators>;
+  const reducer = args.pop() as OnReducer<any, Creators>;
```

Comment:
```suggestion
  const reducer = args.pop() as OnReducer<unknown, Creators>;
```
Would that still work? 🙂